src/main/java/org/seqdoop/hadoop_bam/util/NIOFileUtil.java

```diff
@@ -11,14 +13,36 @@
 import java.nio.file.SimpleFileVisitor;
 import java.nio.file.attribute.BasicFileAttributes;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.stream.Collectors;
 
-class NIOFileUtil {
+public class NIOFileUtil {
   private NIOFileUtil() {
   }
 
-  public static final String PARTS_GLOB = "glob:**/part-[mr]-[0-9][0-9][0-9][0-9][0-9]*";
+  static final String PARTS_GLOB = "glob:**/part-[mr]-[0-9][0-9][0-9][0-9][0-9]*";
+
+  /**
+   * Convert the given path {@link URI} to a {@link Path} object.
+   * @param uri the path to convert
+   * @return a {@link Path} object
+   */
+  public static Path asPath(URI uri) {
+    try {
+      return Paths.get(uri);
+    } catch (FileSystemNotFoundException e) {
+      ClassLoader cl = Thread.currentThread().getContextClassLoader();
+      if (cl == null) {
+        throw e;
+      }
+      try {
+        return FileSystems.newFileSystem(uri, new HashMap<>(), cl).provider().getPath(uri);
```

Comment:
Is there any way to write a for this code path ? Its not obvious to me since it seems like it would require a reference to a FS that isn't instantiated ?